### ISSUE TITLE
swss build fails if SAI library installed

### ISSUE THREAD

**Issue by daall** [Azure/sonic-swss#1122]
configure.ac only generates a Makefile for `tests/mock_tests` if no SAI library is installed:
```
# If no SAI library is installed, compile with SAIVS and run unit tests
AM_COND_IF([HAVE_SAI],[],
           [AC_CONFIG_FILES([tests/mock_tests/Makefile])])
```

However, the Makefile.am for `tests` specifies that `mock_tests` is to be built unconditionally:
```
SUBDIRS = mock_tests
```

As a result, when trying to build swss against a SAI library, the `tests` Makefile attempts to build `mock_tests` first, causing the build to crash.

This was introduced by #1103. We can't revert this commit because it includes a bugfix for portsorch, so we'll need to open another PR to patch this before we can update the swss submodule in sonic-buildimage.

**Comment by stepanblyschak** [comment]
Please check https://github.com/Azure/sonic-swss/pull/1123
